kedro/io/core.py

```diff
@@ -286,6 +336,20 @@ def __init_subclass__(cls, **kwargs: Any) -> None:
         If `_load` or `_save` are defined, alias them as a prerequisite.
 
         """
+
+        init_func: Callable = cls.__init__
+
+        def init_decorator(previous_init: Callable) -> Callable:
+            def new_init(self, *args, **kwargs) -> None:  # type: ignore[no-untyped-def]
+                previous_init(self, *args, **kwargs)
+                if type(self) is cls:
+                    call_args = getcallargs(init_func, self, *args, **kwargs)
+                    self.__post_init__(call_args)
+
+            return new_init
+
+        cls.__init__ = init_decorator(cls.__init__)  # type: ignore[method-assign]
```

Comment:
Worth adding a comment explaining what this does and why is it here?

Reply:
This is in TODO 👍

Reply:
Done